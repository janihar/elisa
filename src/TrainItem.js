import React, { useState, useEffect } from "react";
import { StyleSheet, View, Dimensions, Text, Button } from "react-native";

const TrainItem = props => {
  return (
    <View style={styles.buttonContainer}>
      <Button
        color="#64B82C"
        title={props.trainNumber.toString()}
        onPress={() => alert(props.trainNumber)}
      >
        {props.trainNumber.toString()}
      </Button>
    </View>
  );
};

export default TrainItem;

const styles = StyleSheet.create({
  container: {
    backgroundColor: "#fff",
    alignItems: "center"
  },
  button: { backgroundColor: "#64B82C" },
  buttonContainer: {
    flex: 1,
    justifyContent: "center",
    margin: 10
  }
});
