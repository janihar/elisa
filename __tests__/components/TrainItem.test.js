import React from "react";
import renderer from "react-test-renderer";

import TrainItem from "../../src/TrainItem";

describe("<Trainitem {props}/>", () => {
    const trainNumber = 211
  it("renders correctly", () => {
    const tree = renderer.create(<TrainItem trainNumber={trainNumber} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
