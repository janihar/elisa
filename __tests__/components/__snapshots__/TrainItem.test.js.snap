// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Trainitem {props}/> renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
      "margin": 10,
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#64B82C",
            },
          ]
        }
      >
        211
      </Text>
    </View>
  </View>
</View>
`;
